src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Filters/IDependenciesSnapshotFilter.cs

```diff
@@ -42,16 +44,16 @@ internal interface IDependenciesSnapshotFilter
             out bool filterAnyChanges);
 
         /// <summary>
-        /// Is called before removing a given dependency.
+        /// Called before removing a dependency from a snapshot.
         /// </summary>
         /// <param name="projectPath">Path to current project.</param>
         /// <param name="targetFramework">Target framework for which dependency was resolved.</param>
         /// <param name="dependency">The dependency to which filter should be applied.</param>
         /// <param name="worldBuilder">Builder for immutable world dictionary of updating snapshot.</param>
         /// <param name="topLevelBuilder">Top level dependencies list builder of updating snapshot.</param>
-        /// <param name="filterAnyChanges">True if filter did any changes in the snapshot</param>
-        /// <returns>Dependency itself if removal is approved, null if dependency should not be removed</returns>
-        IDependency BeforeRemove(
+        /// <param name="filterAnyChanges"><see langword="true"/> if <paramref name="worldBuilder"/> or <paramref name="topLevelBuilder"/> changed, otherwise <see langword="false"/>.</param>
+        /// <returns><see langword="true"/> if removal is approved, or <see langword="false"/> if <paramref name="dependency"/> should not be removed.</returns>
+        bool BeforeRemove(
```

Comment:
Would `CanRemove` make more sense as a name? I get that `BeforeRemove` is supposed to imply that the filter could modify things before removal but seems like a thing we don't want to happen anyway.

Reply:
Agreed, CanRemove, ShouldExclude, Filter, IsIgnorable, IsExcludable

Reply:
These "filters" are (currently) intended to observe dependencies being added and removed in order to make other state consistent. If we keep that behaviour, then `Filter` is the misleading bit.